### DISCUSSION TITLE
It is impossible to import all routes from bundle controllers with the new API (AbstractBundle)

### DISCUSSION THREAD

**Discussion by tebaly** [symfony/symfony#51732] [Q&A] [answered]
### Symfony version(s) affected

6.3.0

### Description

Prior to Symfony 6, it was possible to import all controllers from a bundle using the following configuration:
```
# config/routes/acme.yaml
acme:
    resource: '@AcmeBundle/Controller/'
    type: attribute
```
The bundle itself inherited from `Symfony\Component\HttpKernel\Bundle\Bundle`.

In Symfony 6, it is recommended to inherit from `Symfony\Component\HttpKernel\Bundle\AbstractBundle`. 
 
When I simply change Symfony\Component\HttpKernel\Bundle\Bundle to Symfony\Component\HttpKernel\Bundle\AbstractBundle, an error occurs:

> Unable to find file "@AcmeBundle/Controller/" in @AcmeBundle/Controller/ (which is being imported from "/home/app/config/routes/acme.yaml"). Make sure the "AcmeBundle" bundle is correctly registered and loaded in the application kernel class. If the bundle is registered, make sure the bundle path "@AcmeBundle/Controller/" is not empty.

If I revert the inheritance back to `Symfony\Component\HttpKernel\Bundle\Bundle`, everything works as expected without errors.

---

How to use importing all routes from all controllers in a bundle folder Symfony 6+ with a new api AbstractBundle ?

### How to reproduce

simply change Symfony\Component\HttpKernel\Bundle\Bundle to Symfony\Component\HttpKernel\Bundle\AbstractBundle

**Comment by stof** [comment, marked as answer]
AbstractBundle assumes the modern bundle structure by default (when you don't override `getPath()`), so it would look like `@AcmeBundle/src/Controller/` for such bundles.
And if your bundle does not follow the modern structure, you need to override the `getPath` method to make it return the right path

**Comment by tebaly** [comment]
Yes, it works. Unfortunately it wasn't obvious to me

**Reply by stof** [reply]
Well, `@AcmeBundle` is meant to refer to the bundle path. In the new structure, the bundle path is the root of the package.

**Comment by tebaly** [comment]
> Well, `@AcmeBundle` is meant to refer to the bundle path. In the new structure, the bundle path is the root of the package.

What confused me is that with the new structure and compatible API(Symfony\Component\HttpKernel\Bundle\Bundle) this link works `@AcmeBundle/Controller/` And if you just change the inheritance to `Symfony\Component\HttpKernel\Bundle\AbstractBundle` and don’t change anything else, then it stops working.
